operator-common/src/main/java/io/strimzi/operator/common/operator/resource/ConfigParameter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.operator.resource;
+
+import io.strimzi.operator.common.InvalidConfigurationException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Models a configuration parameter, identified by a unique key, which may be required, and if not may have a default value.
+ * Optional parameters without a default value implicitly have a null default.
+ * The key is also the name of the environment variable from which the value may be read, when it is read from the environment.
+ *
+ * @param key Configuration parameter name/key
+ * @param <T> Type of object
+ * @param type type of the default value
+ * @param defaultValue default value of the configuration parameter
+ * @param required  If the value is required or not
+ */
+public record ConfigParameter<T>(String key, AbstractConfig<T> type, String defaultValue, boolean required, Map<String, ConfigParameter<?>> map) {
+    /**
+     * Contructor
+     * @param key Configuration parameter name/key
+     * @param type type of the default value
+     * @param defaultValue default value of the configuration parameter
+     * @param required  If the value is required or not
+     * @param map Configuration map
+     */
+    public ConfigParameter {
+        map.put(key, this);
+    }
+
+    /**
+     * Generates the configuration map
+     * @param envVarMap    Map containing values entered by user.
+     * @param configValues Map containing all the configuration keys with default values
+     * @return             Generated configuration map
+     */
+    public static Map<String, Object> define(Map<String, String> envVarMap, Map<String, ConfigParameter<?>> configValues) {
+
+        Map<String, Object> generatedMap = new HashMap<>(envVarMap.size());
+        for (Map.Entry<String, String> entry : envVarMap.entrySet()) {
+            final ConfigParameter<?> configValue = configValues.get(entry.getKey());
+            generatedMap.put(configValue.key(), get(envVarMap, configValues, configValue));
+        }
+
+        // now add all those config (with default value) that weren't in the given map
+        Map<String, ConfigParameter<?>> x = new HashMap<>(configValues);
+        x.keySet().removeAll(envVarMap.keySet());
+        for (ConfigParameter<?> value : x.values()) {
+            generatedMap.put(value.key(), get(envVarMap, configValues, value));
+        }
+        return generatedMap;
+    }
+    private static <T> T get(Map<String, String> map, Map<String, ConfigParameter<?>> configValues, ConfigParameter<T> value) {
+        if (!configValues.containsKey(value.key())) {
+            throw new InvalidConfigurationException("Unknown config value: " + value.key() + " probably needs to be added to Config.configValues");
```

Comment:
It's always worth thinking about who (as in, which persona) you expect to see an exception message, and what will make sense to them. In this case do they know what `Config.configValues` is? I suspect this is a reference to an out of date class. You could correct that, but in general it's a good idea to avoid putting class, field or method names in messages. If you _have to_ use a class name use a class literal so can't get out of sync with the actual code, e.g. `"Blah blah " + Foo.class.getSimpleName() + " blah blah"`.